pkg/synthetictests/duplicated_events.go

```diff
@@ -77,40 +89,40 @@ type knownProblem struct {
 	BZ     string
 }
 
-func testDuplicatedEventForUpgrade(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+func testDuplicatedEventForUpgrade(events monitorapi.Intervals, kubeClientConfig *rest.Config) []*ginkgo.JUnitTestCase {
 	allowedPatterns := []*regexp.Regexp{}
 	allowedPatterns = append(allowedPatterns, allowedRepeatedEventPatterns...)
 	allowedPatterns = append(allowedPatterns, allowedUpgradeRepeatedEventPatterns...)
 
 	evaluator := duplicateEventsEvaluator{
 		allowedRepeatedEventPatterns: allowedPatterns,
-		allowedRepeatedEventFns:      nil,
+		allowedRepeatedEventFns:      allowedRepeatedEventFns,
 		knownRepeatedEventsBugs:      knownEventsBugs,
 	}
 
-	return evaluator.testDuplicatedEvents(events)
+	return evaluator.testDuplicatedEvents(events, kubeClientConfig)
 }
 
-func testDuplicatedEventForStableSystem(events monitorapi.Intervals) []*ginkgo.JUnitTestCase {
+func testDuplicatedEventForStableSystem(events monitorapi.Intervals, kubeClientConfig *rest.Config) []*ginkgo.JUnitTestCase {
 	evaluator := duplicateEventsEvaluator{
 		allowedRepeatedEventPatterns: allowedRepeatedEventPatterns,
-		allowedRepeatedEventFns:      nil,
+		allowedRepeatedEventFns:      allowedRepeatedEventFns,
 		knownRepeatedEventsBugs:      knownEventsBugs,
 	}
 
-	return evaluator.testDuplicatedEvents(events)
+	return evaluator.testDuplicatedEvents(events, kubeClientConfig)
 }
 
 // isRepeatedEventOKFunc takes a monitorEvent as input and returns true if the repeated event is OK.
 // this commonly happens for known bugs and for cases where events are repeated intentionally by tests.
 // the string is the message to display for the failure.
-type isRepeatedEventOKFunc func(monitorEvent monitorapi.EventInterval) (bool, string)
+type isRepeatedEventOKFunc func(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config) bool
```

Comment:
here you're dropping the string response, but still talk about it in the godocs two lines up.  I [still][1] think it would be more convenient as an `error` return.  And the `context.TODO()` in `isConsoleReadinessDuringInstallation` suggests it should take a `ctx context.Context` argument too.

[1]: https://github.com/openshift/origin/pull/26380#discussion_r683889039